diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -947,25 +1048,17 @@ int DifferentialFunctionComparator::cmpFieldAccess(const Function *L,
 /// of the cast.
 int DifferentialFunctionComparator::cmpValues(const Value *L,
                                               const Value *R) const {
-    // Detect casts and use the original value instead when comparing the
-    // control flow only.
-    const User *UL = dyn_cast<User>(L);
-    const User *UR = dyn_cast<User>(R);
-    const User *CL = (UL && isCast(UL)) ? UL : nullptr;
-    const User *CR = (UR && isCast(UR)) ? UR : nullptr;
-
-    if (CL && CR && mayIgnore(CL) && mayIgnore(CR)) {
-        // Both instruction are casts - compare the original values before
-        // the cast
-        return cmpValues(CL->getOperand(0), CR->getOperand(0));
-    } else if (CL && mayIgnore(CL)) {
-        // The left value is a cast - use the original value of it in the
-        // comparison
-        return cmpValues(CL->getOperand(0), R);
-    } else if (CR && mayIgnore(CR)) {
-        // The right value is a cast - use the original value of it in the
-        // comparison
-        return cmpValues(L, CR->getOperand(0));
+    // Use replacement references for ignored values.
+    auto ignoredL = ignoredInstructions.find(L);
+    auto ignoredR = ignoredInstructions.find(R);
+    auto replaceL = (ignoredL == ignoredInstructions.end()) ? nullptr
+                                                            : ignoredL->second;
+    auto replaceR = (ignoredR == ignoredInstructions.end()) ? nullptr
+                                                            : ignoredR->second;
```

Comment:
Implement a new method `Value *getReplacementValue(const Value *)` that returns the `IgnoredInstructions` map entry if it exists for the given parameter and call it from here.
Also, if the given Value is not an instruction, the `getReplacementValue` method should check if it is a skippable bitcast operator and if so, return its first operand.